### PR TITLE
[improve][fn] Support schema in python instance

### LINKED ISSUE #10114: Support schema for python function
**Is your feature request related to a problem? Please describe.**
Currently there is no support for configuring schema in python functions, some users want to use schema in python functions

**Describe the solution you'd like**
Support Configure schema for python function
Reference: 
Implement topic schema search: https://github.com/apache/pulsar/blob/master/pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/TopicSchema.java
Add schema for python consumer: https://github.com/apache/pulsar/blob/master/pulsar-functions/instance/src/main/python/python_instance.py#L157
Add schema for python producer: https://github.com/apache/pulsar/blob/master/pulsar-functions/instance/src/main/python/python_instance.py#L319

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by jiangpengcheng**
Fixes #10114

### Motivation

Python instance doesn't support schema yet

### Modifications

Add schema support for python instance

### Verifying this change

- [x] Make sure that the change passes the CI checks.

- [x] This change is already covered by existing tests, such as *(please describe tests)*.
  - `org.apache.pulsar.tests.integration.functions.PulsarFunctionsTest`

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/jiangpengcheng/pulsar/pull/2

### CODE REVIEW

File: pulsar-functions/instance/src/main/python/python_instance.py [hunk 1]
```diff
@@ -460,3 +481,45 @@ def close(self):
...
+      schema = DEFAULT_SCHEMA
+    elif schema_type.lower() == "string":
+      schema = pulsar.schema.StringSchema()
+    elif schema_type.lower() == "json":
+      record_kclass = self.get_record_class(type_class_name)
+      schema = pulsar.schema.JsonSchema(record_kclass)
+    elif schema_type.lower() == "avro":
+      record_kclass = self.get_record_class(type_class_name)
+      schema = pulsar.schema.AvroSchema(record_kclass, schema_properties)
+    else:  # load custom schema
```
Comment (freeznet) [inline pulsar-functions/instance/src/main/python/python_instance.py:497]: how about other schema types like PROTOBUF, PROTOBUF_NATIVE, and KeyValue?

**Review by freeznet — COMMENTED** [review]
LGTM, could you please add some more test cases cover other schema types, like JSON and AVRO? thanks.

File: pulsar-functions/instance/src/main/python/python_instance.py (hunk 1 above)
Comment (jiangpengcheng) [reply pulsar-functions/instance/src/main/python/python_instance.py:499]: python client only supports avro, json, string and bytes schema for now

I will add todo here

**Review by freeznet — APPROVED** [review]
lgtm, please change the doc label to `doc-required` because this PR may need some changes on https://pulsar.apache.org/docs/2.11.x/functions-develop-schema-registry/

**Review by nlu90 — APPROVED** [review]

**Comment by momo-jun** [comment]
Hi @jiangpengcheng , I'm checking in to follow up with the doc updates. Did you have any plans to update the docs [here](https://pulsar.apache.org/docs/2.11.x/functions-develop-schema-registry/)?

**Comment by eolivelli** [comment]
@nlu90 @freeznet 
While I see the great value of this patch,
I believe that such API changes MUST be discussed on dev@pulsar.apache.org

Even if our rules are that you only need 2 binding reviews, that's not enough for the sake of the community

**Pull request merged**